src/Nethermind/Nethermind.Merge.Plugin/Data/ForkchoiceStateV1.cs

```diff
@@ -10,35 +10,28 @@ namespace Nethermind.Merge.Plugin.Data;
 ///
 /// <seealso cref="https://github.com/ethereum/execution-apis/blob/main/src/engine/specification.md#forkchoicestatev1"/>
 /// </summary>
-public class ForkchoiceStateV1
+public class ForkchoiceStateV1(Hash256 headBlockHash, Hash256 finalizedBlockHash, Hash256 safeBlockHash)
 {
-    public ForkchoiceStateV1(Hash256 headBlockHash, Hash256 finalizedBlockHash, Hash256 safeBlockHash)
-    {
-        HeadBlockHash = headBlockHash;
-        FinalizedBlockHash = finalizedBlockHash;
-        SafeBlockHash = safeBlockHash;
-    }
-
     /// <summary>
     /// Hash of the head of the canonical chain.
     /// </summary>
-    public Hash256 HeadBlockHash { get; set; }
+    public Hash256 HeadBlockHash { get; set; } = headBlockHash;
 
     /// <summary>
     /// Safe block hash of the canonical chain under certain synchrony and honesty assumptions. This value MUST be either equal to or an ancestor of headBlockHash.
     /// </summary>
     /// <remarks>Can be <see cref="Keccak.Zero"/> when transition block is not finalized yet.</remarks>
-    public Hash256 SafeBlockHash { get; set; }
+    public Hash256 SafeBlockHash { get; set; } = safeBlockHash;
 
     /// <summary>
     /// Hash of the most recent finalized block
     /// </summary>
     /// <remarks>Can be <see cref="Keccak.Zero"/> when transition block is not finalized yet.</remarks>
-    public Hash256 FinalizedBlockHash { get; set; }
+    public Hash256 FinalizedBlockHash { get; set; } = finalizedBlockHash;
 
-    public override string ToString() => $"ForkChoice: {HeadBlockHash.ToShortString()}, Safe: {SafeBlockHash.ToShortString()}, Finalized: {FinalizedBlockHash.ToShortString()}";
+    public override string ToString() => $"ForkChoice: {HeadBlockHash}, Safe: {SafeBlockHash}, Finalized: {FinalizedBlockHash}";
     public string ToString(long? headNumber, long? safeNumber, long? finalizedNumber) =>
         headNumber is null || safeNumber is null || finalizedNumber is null
             ? ToString()
-            : $"ForkChoice: {headNumber} ({HeadBlockHash.ToShortString()}), Safe: {safeNumber} ({SafeBlockHash.ToShortString()}), Finalized: {finalizedNumber} ({FinalizedBlockHash.ToShortString()})";
```

Comment:
This is spammy?

Reply:
Changing this to multiple lines

<img width="1027" height="424" alt="Image" src="https://github.com/user-attachments/assets/4373db42-9683-49bf-ad59-ebdd73a70fae" />

Reply:
reverted